src/hooks/useFetch.ts

```diff
@@ -41,55 +63,78 @@ export function useFetch<T>(options: UseFetchOptions) {
     skip = false,
     auto = true,
   } = options;
+
   const [data, setData] = useState<T | null>(null);
+
   const [loading, setLoading] = useState(false);
+
   const [error, setError] = useState<Error | null>(null);
 
   const fetchData = useCallback(
     async (overrideOptions?: Partial<UseFetchOptions>) => {
       setLoading(true);
+
       setError(null);
+
       try {
         const url = getRequestUrl({
           baseUrl: overrideOptions?.baseUrl ?? baseUrl,
           path: overrideOptions?.path ?? path,
           searchParams: overrideOptions?.searchParams ?? searchParams,
         });
+
         const res = await fetch(url, {
           method: overrideOptions?.method ?? method,
+
           headers: overrideOptions?.headers ?? headers,
+
           body: overrideOptions?.body
             ? JSON.stringify(overrideOptions.body)
             : body
               ? JSON.stringify(body)
               : undefined,
         });
+
         if (!res.ok) {
-          throw new Error('@kakao.com 이메일 주소만 가능합니다.');
+          let errData: Record<string, unknown> = {};
+          try {
+            errData = await res.json();
+          } catch {}
+          const errorMsg =
+            typeof errData.message === 'string'
+              ? errData.message
+              : res.statusText;
+          throw new FetchError(errorMsg, res.status, errData);
```

Comment:
[제안] `Response` 객체를 받아 자동으로 FetchError의 생성자 매개변수를 채워주는 헬퍼 메서드로 추상화하면 사용하기 편하겠네요
```suggestion
          throw await FetchError.fromResponse(res);
```